src/ifcopenshell-python/ifcopenshell/api/pset/edit_pset.py

```diff
@@ -244,22 +244,46 @@ def add_new_properties(self):
                 continue
             unit, value = self.unpack_unit_value(value)
             if isinstance(value, ifcopenshell.entity_instance):
-                if value.is_a(True) == "IFC4.IfcPropertyEnumeratedValue":
+                if value.is_a("IfcProperty"):
                     properties.append(value)
-                    continue
-                else:
+
+                elif not value.is_entity():
```

Comment:
@aothms I'm a bit lost in this point. From my ignorance, isn't it possible to do something like value.is_a('IfcValue') for an IfcInteger, IfcLabel, etc.? or this doesn't work for the Resource Layer? I see it throws False. What does `is_entity()` return on an entity_instance?

Reply:
It is unclear what's I'm doing here, I don't know of a better way to check if it's a primitive data type.

`value.is_a("IfcValue")` doesn't work, perhaps because `IfcValue` doesn't actually exist, it's some kind of abstract enumeration/union that holds other abstract enumeration/union that in the end hold actual ifc classes like `IfcLabel`.

`value.is_entity()` indeed works, for entity it returns `True`, for primitives it returns `False`:
https://github.com/IfcOpenShell/IfcOpenShell/blob/032f52d434e13059fd24c12436ec71a645a5108c/src/ifcopenshell-python/ifcopenshell/entity_instance.py#L342-L352

If there's a more clear way of doing this, let me know.

Reply:
> value.is_a("IfcValue") doesn't work

If we agree that it *should* work, then I don't mind adding support for this. It's related to #3187

Reply:
I think that'd be useful, but I don't know if it'd case any overhead.